db/httpdb.js

```diff
@@ -439,6 +440,52 @@ module.exports = function (
     )
   }
 
+  DB.prototype.lockAccount = function (account) {
+    var unlockCode = crypto.randomBytes(16).toString('hex');
+    log.trace({ op: 'DB.lockAccount', uid: account && account.uid, unlockCode: unlockCode })
+
+    return this.pool.post(
+      '/account/' + account.uid.toString('hex') + '/lock',
+      {
+        lockedAt: Date.now(),
+        unlockCode: unlockCode
+      }
+    )
+  }
+
+  DB.prototype.unlockAccount = function (account) {
+    log.trace({ op: 'DB.unlockAccount', uid: account && account.uid })
+    return this.pool.post(
+      '/account/' + account.uid.toString('hex') + '/unlock'
+      )
+      .then(null,
+        function (err) {
+          if (err.statusCode === 404) {
+            err = error.unknownAccount()
```

Comment:
@chilts & @rfk - I can only get this case to occur with the mem db when the account does not exist. The MySQL happily succeeds. Whether an existing account is actually locked is irrelevant to both. Filed https://github.com/mozilla/fxa-auth-db-mem/issues/16 and https://github.com/mozilla/fxa-auth-db-mysql/issues/11 about the inconsistency w/ non-existent accounts.